h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -149,20 +61,17 @@ public static AutoML startAutoML(AutoMLBuildSpec buildSpec) {
     synchronized (AutoML.class) {
       // protect against two runs whose startTime is the same second
       if (lastStartTime != null) {
-        while (lastStartTime.getYear() == startTime.getYear() &&
-            lastStartTime.getMonth() == startTime.getMonth() &&
-            lastStartTime.getDate() == startTime.getDate() &&
-            lastStartTime.getHours() == startTime.getHours() &&
-            lastStartTime.getMinutes() == startTime.getMinutes() &&
-            lastStartTime.getSeconds() == startTime.getSeconds())
+        while (Math.abs(lastStartTime.getTime() - startTime.getTime()) < 1e3 ) {
+          try {
+            Thread.sleep(500);
```

Comment:
I've also been thinking about tricking the Date instance indeed.
Honestly, I don't think anyone creates AutoML instances in a loop anyway, as it would hurt performance: H2O-3 doesn't scale in those situations as model building is trying to use all the nodes/threads (even if it doesn't always).
Also, I know that the loop could check more often, but given that the AutoML run will last several minutes (hours?), who can't wait a second more?
Still, maybe the trick with the Date would be nicer than this.

Reply:
@seb-h2o sorry, obviously I was not clear enough in explanation :) I was talking about reducing 500ms to a smaller value ( and making loops shorter) let's say 10ms or 50 ms - it will not hurt and just reduce maximum of possible waste of time. Agree, it is not a big difference after all but if we can easily mitigate this loss then maybe it is also not a big deal.